### DISCUSSION TITLE
Error message assertions

### DISCUSSION THREAD

**Discussion by sjamesr** [WebAssembly/spec#1838] [Q&A] [answered]
Hi all,

I am working on a spec test runner for various engines. Are compliant engines required to produce trap/exception messages that exactly match the spec test? For example, in `bulk.wast`, there is an `assert_trap` that asserts the engine emits the index of an uninitialized memory element, e.g. `uninitialized element 2`. Is an engine that produces `uninitialized element` compliant? Is it sufficient that the engine traps, or must it produce exactly this message?

**Comment by rossberg** [comment, marked as answer]
The messages merely have informational character. There is no normative requirement to replicate them, or even produce messages at all.
